koassets-react/src/components/Picture.tsx

```diff
@@ -0,0 +1,52 @@
+import type { Asset } from '../types';
+import React, { useState } from 'react';
+
+interface PictureProps {
+    asset: Asset;
+    width: number;
+    className?: string;
+}
+
+const Picture: React.FC<PictureProps> = ({
+    asset,
+    width,
+    className = ''
+}) => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [hasError, setHasError] = useState(false);
+
+    const id = asset.assetId || '';
+    const encodedId = encodeURIComponent(id);
```

Comment:
Is this necessary? AFAIK the asset ids are ok to use as is in URLs. Haven't seen a problem with them...

Reply:
I don't think it is. The uuid from AEM is already url safe.

Note that the [DM API docs](https://developer.adobe.com/experience-cloud/experience-manager-apis/api/stable/assets/delivery/#operation/getAssetSeoFormat) for `assetId` don't mention this, unlike say for the `seoName`:
<img width="1034" height="368" alt="Bildschirmfoto 2025-10-06 um 12 51 50" src="https://github.com/user-attachments/assets/cb5fccb2-8b6e-4405-ae92-5d2505d54bc2" />